indy_node/server/request_handlers/read_req_handlers/get_attribute_handler.py

```diff
@@ -1,74 +1,109 @@
-from indy_node.server.request_handlers.domain_req_handlers.attribute_handler import AttributeHandler
-from plenum.server.request_handlers.handler_interfaces.read_request_handler import ReadRequestHandler
-from indy_common.constants import ATTRIB, GET_ATTR
-from indy_node.server.request_handlers.utils import validate_attrib_keys
-from plenum.common.constants import RAW, ENC, HASH, TARGET_NYM, DOMAIN_LEDGER_ID
+from typing import Any, Mapping
+
+from plenum.common.constants import DOMAIN_LEDGER_ID, ENC, HASH, RAW, TARGET_NYM
 from plenum.common.exceptions import InvalidClientRequest
 from plenum.common.request import Request
 from plenum.common.txn_util import get_request_data
 from plenum.server.database_manager import DatabaseManager
+from plenum.server.request_handlers.utils import decode_state_value
 from stp_core.common.log import getlogger
 
+from indy_common.constants import ATTRIB, GET_ATTR, VERSION_ID, VERSION_TIME
+from indy_node.server.request_handlers.domain_req_handlers.attribute_handler import (
+    AttributeHandler,
+)
+from indy_node.server.request_handlers.read_req_handlers.version_read_request_handler import (
+    VersionReadRequestHandler,
+)
+from indy_node.server.request_handlers.utils import validate_attrib_keys
+
 logger = getlogger()
 
 
-class GetAttributeHandler(ReadRequestHandler):
+class GetAttributeHandler(VersionReadRequestHandler):
 
-    def __init__(self, database_manager: DatabaseManager):
-        super().__init__(database_manager, GET_ATTR, DOMAIN_LEDGER_ID)
+    def __init__(self, node, database_manager: DatabaseManager):
+        super().__init__(node, database_manager, GET_ATTR, DOMAIN_LEDGER_ID)
 
     def get_result(self, request: Request):
         self._validate_request_type(request)
+
         identifier, req_id, operation = get_request_data(request)
+
+        timestamp = operation.get(VERSION_TIME)
+        seq_no = operation.get(VERSION_ID)
+        if timestamp and seq_no:
+            raise InvalidClientRequest(
+                request.identifier,
+                request.reqId,
+                f"{VERSION_ID} and {VERSION_TIME} are mutually exclusive; only one should be "
+                "specified",
+            )
```

Comment:
I see the mutual exclusion test in `VersionReadRequestHandler` as the "am I using this method correctly?" check and the mutual exclusion test in the get nym and attrib handlers as a "is the request valid?" checks. Yes, there's a bit of duplicated code but I think I prefer the clarity over terseness in this case.

Reply:
That makes sense.  It would be worth noting that in the code, so someone does not come along later and remove it because they see the checks as redundant.

Reply:
Comments added!